sumcheck/src/verifier/mod.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+use alloc::vec::Vec;
+
+use air::LogUpGkrEvaluator;
+use crypto::{ElementHasher, RandomCoin};
+use math::FieldElement;
+
+use crate::{
+    evaluate_composition_poly, EqFunction, FinalLayerProof, FinalOpeningClaim, RoundProof,
+    SumCheckProof, SumCheckRoundClaim,
+};
+
+/// Verifies a round of the sum-check protocol without executing the final check.
+pub fn verify_rounds<E, C, H>(
+    claim: E,
+    round_proofs: &[RoundProof<E>],
+    coin: &mut C,
+) -> Result<SumCheckRoundClaim<E>, SumCheckVerifierError>
+where
```

Comment:
Does this need to be public? It is used in the other two functions here - but I'm not sure if it also needs to be used individually.

Also nit re using `&mut impl RandomCoin<Hasher = H, BaseField = E::BaseField>` instead of `C` type parameter.

Reply:
Changed